src/Generating/FinishGen.cpp

```diff
@@ -801,6 +802,11 @@ void cFinishGenSprinkleFoliage::GenFinish(cChunkDesc & a_ChunkDesc)
 					)
 					{
 						a_ChunkDesc.SetBlockType(x, ++Top, z, E_BLOCK_CACTUS);
+
+						for (int i = 0; i < (x % m_MaxCactusHeight); i++)
+						{
+							a_ChunkDesc.SetBlockType(x, ++Top, z, E_BLOCK_CACTUS);
```

Comment:
This code is a bit confusing, could you use `Top + i + 1` instead?

Reply:
Yup, i'll replace `++Top`. Apparently, it's also the case for sugar canes - I'll include a fix for it too